pydicom/tests/test_valuerep.py

```diff
@@ -889,11 +891,24 @@ def test_valid_value(self, disable_value_validation):
         assert 42 == IS("42.0")
         assert 42 == IS(42.0)
 
+    def test_float_value(self):
+        """Read binary value of IS that is actually a float"""
+        # from issue #1661
+        # Create BytesIO with single data element for Exposure (0018,1152)
+        #   length 4, value "14.5"
+        bin_elem = b"\x18\x00\x52\x11\x04\x00\x00\x0014.5"
+        with BytesIO(bin_elem) as bio:
+            ds = read_dataset(bio, True, True)
+        assert isinstance(ds.Exposure, ISfloat)
```

Comment:
Not strictly necessary to go back to binary like this, but I thought it a good idea in case of future hooks into reading invalid values.

Reply:
Agreed!